hazelcast/src/main/java/com/hazelcast/function/SupplierEx.java

```diff
@@ -47,4 +47,8 @@ default T get() {
         }
     }
 
+    static <T> SupplierEx<T> noop() {
+        return () -> null;
```

Comment:
checking for null is not an issue here, that is the standard way of deciding if an operation requires a permission or not.

how about this:
```
private Permission requiredPermission;
```
the permission is serializable and immutable, we can directly ask for it instead of a supplier?

Reply:
yes, that would work